core/src/View/Kokkos_ViewLegacy.hpp

```diff
@@ -246,12 +246,19 @@ class View : public ViewTraits<DataType, Properties...> {
 
  public:
   //----------------------------------------
-  /** \brief  Compatible view of array of scalar types */
-  using array_type =
+#ifdef KOKKOS_ENABLE_DEPRECATED_CODE_4
+  /** \brief  Compatible view of array of data types */
+  using array_type KOKKOS_DEPRECATED_WITH_COMMENT("Use type instead.") =
       View<typename traits::scalar_array_type, typename traits::array_layout,
            typename traits::device_type, typename traits::hooks_policy,
            typename traits::memory_traits>;
+#endif
 
+  KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_PUSH()
```

Comment:
What warnings are we ignoring?!
I am surprised this is needed.

Reply:
In this block we're ignoring the warnings that come from using ViewTraits that include the deprecated stuff when defining the View type in L259.

Reply:
That said, I agree that the macros should ideally be in the unit tests and not here. I still see CI test errors (https://cloud.cees.ornl.gov/jenkins-ci/job/Kokkos/job/PR-7360/18/pipeline-graph/) coming from the unit tests, any idea why they still occur? I feel like I've added the right macros.